src/kili/helpers.py

```diff
@@ -374,3 +385,96 @@ def __call__(self, retry_state: tenacity.RetryCallState):
         if not self.warned and float(retry_state.outcome_timestamp or 0) > self.warn_after:
             self.logger_func(self.warn_message)
             self.warned = True
+
+
+D = TypeVar("D")
+
+
+def skip_if_empty_arguments(
+    all_non_empty: Optional[Sequence[str]] = None, any_non_empty: Optional[Sequence[str]] = None
+) -> Callable[[Callable[..., D]], Callable[..., Optional[D]]]:
+    """
+    Decorator that warns if an argument is an empty sequence.
+
+    If a warning is raised, the decorated method is skipped and returns None.
+
+    Args:
+        all_non_empty: sequence of argument names that must be all non-empty.
+        any_non_empty: sequence of argument names that must have at least one non-empty value.
+    """
+
+    def is_empty_sequence(arg_value) -> bool:
+        return isinstance(arg_value, Sequence) and len(arg_value) == 0
+
+    def get_arg_value(arg_name: str, args: Tuple[Any, ...], kwargs: Dict[str, Any], func: Callable):
+        """
+        Get the value of an argument `arg_name` from the args and kwargs of the decorated function.
+        """
+        if arg_name in kwargs:
+            return kwargs[arg_name]
+
+        index_ = func.__code__.co_varnames.index(arg_name)
+        try:
+            return args[index_]
+        except IndexError:
+            return None
+
+    def get_args_as_dict(
+        arg_names: Sequence[str], args: Tuple[Any, ...], kwargs: Dict[str, Any], func: Callable
+    ) -> Dict[str, Any]:
+        args_dict = {}
+        for arg_name in arg_names:
+            args_value = get_arg_value(arg_name, args, kwargs, func)
+            if args_value is not None:
+                args_dict[arg_name] = args_value
+        return args_dict
+
+    def decorator(func: Callable[..., D]) -> Callable[..., Optional[D]]:
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs) -> Optional[D]:
```

Comment:
alternatively this could return an empty object of type D, to not change the return type. But I guess that's more complicated to implement.

Reply:
If we have a method that returns a {"id": "123456"} for example, you would return an empty dict?  I think returning None is fine

Reply:
good point